File: Basic Requests + Playwright/Rotating Residential/simpleRequest.js
import fetch from 'node-fetch';
import { HttpsProxyAgent } from 'https-proxy-agent';
import dotenv from 'dotenv';

// Load environment variables from .env file
dotenv.config();

// Access environment variables
const userName = process.env.USERNAME;
const password = process.env.PASS;

// Proxy configuration
const proxyAgent = new HttpsProxyAgent(
  `http://${userName}:${password}@proxy.proxy-cheap.com:31112`
);

// Making the request
fetch('https://ipv4.icanhazip.com', { agent: proxyAgent })
  .then((response) => response.text())
  .then((text) => console.log(text))
  .catch((error) => console.error('Error:', error));

File: Basic Requests + Playwright/Rotating Residential/simple_request.py
import requests
import os
from dotenv import load_dotenv

load_dotenv()

user_name = os.getenv("USERNAME")
# user_name = "w4g76yre"
password = os.getenv("PASS")
print(user_name, password)
# Proxy settings
proxies = {
    "http": f"http://{user_name}:{password}@proxy.proxy-cheap.com:31112",
    "https": f"http://{user_name}:{password}@proxy.proxy-cheap.com:31112",
}

# Make the request
response = requests.get("https://ipv4.icanhazip.com", proxies=proxies)

# Print the response text (IP address)
print(response.text.strip())
